### PR TITLE
Remove logger-initialization in __init__.py

### PR DESCRIPTION
**Pull request by rubencalje**
# Short Description
This PR removes the logger initialization when importing Pastas. See issue #733 for more details. This will change the messages people receive in their console, as info-messages will not be displayed by default anymore. People can use the following lines in their script to show info-messages:

```
import logging
logging.basicConfig(level=logging.INFO)
```

This PR removes the setting of any logger handlers from Pastas, because of the follwing note in https://docs.python.org/3/howto/logging.html#configuring-logging-for-a-library: 

> It is strongly advised that you do not add any handlers other than NullHandler to your library’s loggers. This is because the configuration of handlers is the prerogative of the application developer who uses your library. The application developer knows their target audience and what handlers are most appropriate for their application: if you add handlers ‘under the hood’, you might well interfere with their ability to carry out unit tests and deliver logs which suit their requirements.

# Checklist before PR can be merged:
- [x] closes issue #733
- [x] is documented
- [x] Format code with [Black formatting](https://black.readthedocs.io)
- [ ] type hints for functions and methods
- [ ] tests added / passed
- [ ] Example Notebook (for new features)
- [ ] Remove output for all notebooks with changes

### CODE REVIEW

**Comment by rubencalje** [comment]
I do not understand the ruff-errors. This PR is ready.

File: doc/examples/example_logger.py
Comment (raoulcollenteur) [inline doc/examples/example_logger.py:24]: Should we not deprecate this method then?

**Review by raoulcollenteur — CHANGES_REQUESTED** [review]

**Comment by raoulcollenteur** [comment]
PR looks good. One thing before merging. 

ps.set_log_level method is basically rendered useless, no? I perhaps I misunderstand. But if it is not useful anymore we could remove it from the examples and deprecate it?

File: doc/examples/example_logger.py
Comment (rubencalje) [reply doc/examples/example_logger.py:24]: The set_log_level-method is useful, as it sets the log-level of Pastas. The messages are still not printed, though, as we have not initialized a logger yet (which is what we do in the next line).

**Comment by rubencalje** [comment]
> PR looks good. One thing before merging.
> 
> ps.set_log_level method is basically rendered useless, no? I perhaps I misunderstand. But if it is not useful anymore we could remove it from the examples and deprecate it?

ps.set_log_level is still usefull, so I am in favor of keeping it.

**Comment by raoulcollenteur** [comment]
RUFF was reporting a missing line between imports :p Oh well... I'll merge after the checks are done.

**Review by raoulcollenteur — APPROVED** [review]

**Pull request merged**
